### PR TITLE
dev: Get the block status from the actual block in get_block_with_tx_hashes

### PR DESCRIPTION
**Pull request by azurwastaken**
# Pull Request type

Please add the labels corresponding to the type of changes your PR introduces:

- Feature

## What is the current behavior?

Currently the block status is hardcoded with Accepted_on_L2

Resolves: #1369

## What is the new behavior?

- If the last block number is superior or equal to retrieved block number, we set the status to Accepted_on_L1 
- Otherwise, we know the block so its Accepted on L2

## Does this introduce a breaking change?

No

### CODE REVIEW

File: crates/client/rpc/src/lib.rs [hunk 1]
```diff
@@ -847,12 +848,21 @@ where
...
         };
+        let block_status = match self.backend.messaging().last_synced_l1_block_with_event() {
+            Ok(l1_block) => {
+                if l1_block.block_number >= starknet_block.header().block_number {
+                    BlockStatus::AcceptedOnL1
+                } else {
+                    BlockStatus::AcceptedOnL2
+                }
+            },
+            Err(e) => Err(anyhow!("couldn't retrieve block from l1 : {}", e)) ?,
```
Comment (EvolveArt) [inline crates/client/rpc/src/lib.rs:859]: we should probably do the same as above here i.e return a CallError::Failed
```
CallError::Failed(anyhow::anyhow!(
                        "The hash cached for block with hash {block_hash:?} is an invalid felt: '{tx_hash}'. The \
                         caching db has probably been tempered"
                    ))
                
```

**Review by EvolveArt — CHANGES_REQUESTED** [review]

File: crates/client/rpc/src/lib.rs (hunk 1 above)
Comment (azurwastaken) [reply crates/client/rpc/src/lib.rs]: done

**Review by EvolveArt — APPROVED** [review]

File: crates/client/rpc/src/lib.rs [hunk 2]
```diff
@@ -847,12 +847,21 @@ where
...
         };
+        let block_status = match self.backend.messaging().last_synced_l1_block_with_event() {
+            Ok(l1_block) => {
+                if l1_block.block_number >= starknet_block.header().block_number {
+                    BlockStatus::AcceptedOnL1
+                } else {
+                    BlockStatus::AcceptedOnL2
+                }
+            }
+            Err(e) => Err(CallError::Failed(anyhow::anyhow!("couldn't retrieve block from l1 : {}", e)))?,
```
Comment (tdelabro) [inline crates/client/rpc/src/lib.rs:858]: return `StarknetRpcApiError::InternalServerError` and log the DbError you got from `last_synced_l1_block_with_event`

**Review by tdelabro — CHANGES_REQUESTED** [review]
good job

File: crates/client/rpc/src/lib.rs (hunk 2 above)
Comment (azurwastaken) [reply crates/client/rpc/src/lib.rs]: modified :)

**Review by tdelabro — APPROVED** [review]

**Pull request merged**